### ISSUE TITLE
large-hashable-0.1.0.3 build fails on x86 platform

### ISSUE THREAD

**Issue by peti** [factisresearch/large-hashable#11]
Citing from https://build.opensuse.org/package/live_build_log/devel:languages:haskell:lts:8/ghc-large-hashable/openSUSE_Tumbleweed/i586:

~~~~
[   69s] src/Data/LargeHashable/MD5.hs:97:20: error:
[   69s]     • Couldn't match expected type ‘Word32’ with actual type ‘Word64’
[   69s]     • In the first argument of ‘f’, namely ‘(w128_first h)’
[   69s]       In a stmt of a 'do' block: f (w128_first h)
[   69s]       In the expression:
[   69s]         do { f (w128_first h);
[   69s]              f (w128_second h) }
[   69s] 
[   69s] src/Data/LargeHashable/MD5.hs:98:20: error:
[   69s]     • Couldn't match expected type ‘Word32’ with actual type ‘Word64’
[   69s]     • In the first argument of ‘f’, namely ‘(w128_second h)’
[   69s]       In a stmt of a 'do' block: f (w128_second h)
[   69s]       In the expression:
[   69s]         do { f (w128_first h);
[   69s]              f (w128_second h) }
[   69s] [4 of 6] Compiling Data.LargeHashable.Class ( src/Data/LargeHashable/Class.hs, dist/build/Data/LargeHashable/Class.o )
[   69s] 
[   69s] src/Data/LargeHashable/Class.hs:177:18: error:
[   69s]     • Couldn't match type ‘Int64’ with ‘Int32’
[   69s]       Expected type: Int64 -> LH ()
[   69s]         Actual type: Int32 -> LH ()
[   69s]     • In the expression: updateHashWithFun hu_updateLong . CLong
[   69s]       In an equation for ‘updateHash’:
[   69s]           updateHash = updateHashWithFun hu_updateLong . CLong
[   69s]       In the instance declaration for ‘LargeHashable Int64’
[   69s] 
[   69s] src/Data/LargeHashable/Class.hs:192:18: error:
[   69s]     • Couldn't match type ‘Word64’ with ‘Word32’
[   69s]       Expected type: Word64 -> LH ()
[   69s]         Actual type: Word32 -> LH ()
[   69s]     • In the expression: updateHashWithFun hu_updateULong . CULong
[   69s]       In an equation for ‘updateHash’:
[   69s]           updateHash = updateHashWithFun hu_updateULong . CULong
[   69s]       In the instance declaration for ‘LargeHashable Word64’
[   69s] error: Bad exit status from /var/tmp/rpm-tmp.gg9nkp (%build)
~~~~~

**Comment by peti** [comment]
Ping?

**Comment by sternenseemann** [comment]
I think, I have identified the problem and am working on a PR fixing this.

**Comment by sternenseemann** [comment]
Should be fine as soon as #12 is merged and released.

**Comment by skogsbaer** [comment]
fixed in 0.1.04
